tests/test_types.py

```diff
@@ -630,3 +631,21 @@ def test_deepcopy_of_singleton_fixed_type() -> None:
 
     for lhs, rhs in zip(list_of_fixed_types, copied_list):
         assert id(lhs) == id(rhs)
+
+
+def test_strtobool() -> None:
+    # Values that should return True
+    true_values = ["y", "yes", "t", "true", "on", "1"]
+    for val in true_values:
+        assert strtobool(val) == 1, f"Expected True for value: {val}"
```

Comment:
nit 
```
assert strtobool(val) is True
```

Reply:
The ruff formatter during linting is causing trouble when I am comparing with True/False instead of 0/1 in the unit test hence the difference, But it should not affect as This behaviour is based on the truthiness of the values.
0 is equivalent to False and 1 is equivalent to True.

Reply:
```
assert strtobool(val) is True, f"Expected True for value: {val}"
```

and `make lint` works for me

Reply:
what is the error youre getting?

Reply:
<img width="992" alt="image" src="https://github.com/apache/iceberg-python/assets/66407733/e355e96b-6730-4f1d-869d-d52ed8ee370a">

Reply:
nevermind, using `is` instead of `==` fixed the issue.